File: OpenConvo/Assets/Editor/OpenConvo/OCTreeInspector.js
#pragma strict

import System.Collections.Generic;

@CustomEditor ( OCTree )
public class OCTreeInspector extends Editor {
	override function OnInspectorGUI () {
		var tree : OCTree = target as OCTree;

		if ( GUILayout.Button ( "Open editor", GUILayout.Height ( 32 ) ) ) {
			OCTreeEditor.target = tree;
			EditorWindow.GetWindow ( OCTreeEditor );
		}
	}
}

File: OpenConvo/Assets/Editor/OpenConvo/OCTreeInspector.cs
using UnityEngine;
using UnityEditor;
using System.Collections;
using System.Collections.Generic;

[CustomEditor (typeof(OCTree))]
public class OCTreeInspector : Editor {
	override public void OnInspectorGUI () {
		OCTree tree = (OCTree) target;

		List< string > roots = new List< string > ();

		for ( int i = 0; i < tree.rootNodes.Length; i++ ) {
			roots.Add ( i.ToString() );
		}

		tree.currentRoot = EditorGUILayout.Popup ( "Current root", tree.currentRoot, roots.ToArray() );

		if ( GUILayout.Button ( "Open editor", GUILayout.Height ( 32 ) ) ) {
			OCTreeEditor.target = tree;
			EditorWindow.GetWindow ( typeof(OCTreeEditor) );
		}
	}
}
